music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,348 @@
+                                             inString=measureString,  # Creating update
+                                             )
+
+            self.combinedList.append(measureString)
+
+    def getChordString(self, rn):
+        '''
+        Produce a string from a Roman number with the chord and
+        the key if that key constitutes a change from the foregoing context.
+        '''
+
+        keyString = str(rn.key).split(' ')[0].replace('-', 'b')
+        if keyString != self.currentKey:
+            self.currentKey = keyString
+            return f'{keyString}: {rn.figure}'
+        else:
+            return str(rn.figure)
+
+# ------------------------------------------------------------------------------
+
+    # To write
+
+    def writeRomanText(self,
+                       outPath: str = '.',
+                       fileName: str = ''):
+        '''
+        Writes the combined information to a 'Roman text' file (using the extension .txt).
+        '''
+
+        if not self.combinedList:
+            self.prepList()
+
+        f = fileName or self.fileName
+        text_file = open(os.path.join(outPath, f'{f}.txt'), "w")
+        for entry in self.combinedList:
+            text_file.write(entry + "\n")
+        text_file.close()
+
+
+# ------------------------------------------------------------------------------
+
+# Static functions
+
+def rnString(measure,
+             beat,
+             chordString: str,
+             inString: str = ''):
+    '''
+    Writes lines of RNTXT.
+    To start a new line: inString = None;
+    To extend an existing line: set inString to that existing list.
+    '''
+
+    if not inString:  # New line
+        inString = f'm{measure}'
+
+    bt = intBeat(beat)
+
+    newString = f'{inString} b{bt} {chordString}'
+
+    return newString
+
+
+def intBeat(beat,
+            roundValue: int = 2):
+    '''
+    Converts beats to integers if possible, and otherwise to rounded decimals.
+    Accepts input as string, int, float, or fractions.Fraction.
+    '''
+
+    options = [str, int, float, fractions.Fraction]
+
+    if type(beat) not in options:
+        raise TypeError(f'Beat, (currently {beat}) must be one of {options}.')
+
+    if isinstance(beat, int):
+        return beat
+
+    if type(beat) in [str, fractions.Fraction]:
+        beat = float(beat)
+
+    # Now beat is definitely a float
+    if int(beat) == beat:
+        return int(beat)
+    else:
+        return round(beat, roundValue)
+
+
+# ------------------------------------------------------------------------------
+
+class Test(unittest.TestCase):
+    '''
+    Tests for both main analysis cases - one full, one partial - and for a template.
+    Additional test for smaller static functions.
+    '''
+
+    def testTwoCorpusPiecesAndTwoCorruptions(self):
+
+        from music21 import corpus
+        from music21 import meter
+
+        scoreBach = corpus.parse('bach/choraleAnalyses/riemenschneider001.rntxt',
```

Comment:
How much additional time is this parsing adding to the music21 test suite?  I believe that RNTXT parsing is very slow.  There are much smaller test files that could be used.

Reply:
Replaced cases with smallest files in the Bach/Monteverdi collections. Now the test (9 of them including the new ones run in < 0.3 seconds).